### DISCUSSION TITLE
[advice] Pseudo private methods

### DISCUSSION THREAD

**Discussion by SpareShade** [pmndrs/zustand#1277] [Q&A] [answered]
Long story short... as typescript `interface` exposes only public methods you "cannot have" private methods on Zustand.
The workaround is to have non-exported functions as a sort of pseudo private methods.

I'm wondering if there are any shortcomings by passing `SetState<FilesStoreState>` and `GetState<FilesStoreState>` around, eg:

```js
const fetchFiles = async (
  set: SetState<FilesStoreState>,
  get: GetState<FilesStoreState>
) => {
  if (get().pendingFileIdsToFetch.length > 0) {
    const idsToFetch = get().pendingFileIdsToFetch.map((id) => id);
    // empty pendingFileIdsToFetch ... 
    set((s) => {
      s.pendingFileIdsToFetch = [];
    });
    // await fetch files [...idsToFetch]
    console.log("fetching", idsToFetch);
  }
};
``` 

All thoughts much appreciated :)

**Comment by dai-shi** [comment, marked as answer]
By pseudo private, do you mean something TypeScript can detect it? Or, should it be "hard" to use private methods in JS too?

Is external function instead of method acceptable? https://github.com/pmndrs/zustand/blob/main/docs/guides/practice-with-no-store-actions.md

**Reply by SpareShade** [reply]
@dai-shi yes that's it, nice and neat, thank you!

Yes, I meant "hard" to use, in a sense of being a helper method/function for the store, but not being exposed by the store to its clients.

Thank you for sharing this awesome library!!!
